### PR TITLE
Get rid of some hacks in declarations file for webpack-4

### PR DESCRIPTION
**Pull request by jeffin143**
**What kind of change does this PR introduce?**
Backport #10050 to webpack-4

Fix - #10217 
**Did you add tests for your changes?**
NA

**Does this PR introduce a breaking change?**
None

**What needs to be documented once your changes are merged?**
Nothing

### CODE REVIEW

File: declarations.d.ts [hunk 1]
```diff
@@ -228,36 +228,4 @@ declare module "@webassemblyjs/ast" {
...
-	extends Array<
-		import("./declarations/WebpackOptions").RuleSetConditionAbsolute
-	> {}
-
-/**
- * Global variable declarations
- * @todo Once this issue is resolved, remove these globals and add JSDoc onsite instead
- * https://github.com/Microsoft/TypeScript/issues/15626
- */
-declare const $hash$;
```
Comment (sokra) [inline declarations.d.ts]: There are some more changes required to remove this.

In the original PR I added variable declarations in files where these are used.

**Comment by jeffin143** [comment]
@sokra , need help with error :

`declarations/WebpackOptions.d.ts need to be updated`

**Comment by sokra** [comment]
> @sokra , need help with error :
> 
> `declarations/WebpackOptions.d.ts need to be updated`

run `yarn special-lint-fix`

**Comment by jeffin143** [comment]
@sokra should we still keep node-6 pipeline ?

**Comment by sokra** [comment]
Thanks

**Pull request merged**